### PR TITLE
Patch react-native/Yoga to make it possible to build an app using Xcode 14.3

### PR DESCRIPTION
**Pull request by alwx**
The issue was described in #status-mobile channel — Status is not buildable in XCode 14.3 due to an issue with Yoga.cpp — it contains a bitwise | instead of || so it's not buildable. 
One of the solutions is to update to Yoga 1.19.0 but we cannot do that yet (it will probably be done with the update of React Native itself). Instead we can just patch the file using a `postinstall` hook.

#### Platforms

- iOS

status: ready

### CODE REVIEW

**Comment by jakubgs** [comment]
We should avoid including extra dependencies just to apply a patch.

We can do it in Nix already without need for extra JS dependencies:
https://github.com/status-im/status-mobile/blob/develop/nix/deps/nodejs-patched/default.nix

**Comment by jakubgs** [comment]
Also, this fails because you updated `package.json` but did not update `yarn.lock`, so it cannot work:
https://github.com/status-im/status-mobile/blob/4123fde91bcb91118c72385cc6318abcd30d7799/nix/deps/nodejs/default.nix#L4-L10

**Comment by alwx** [comment]
I've updated it to use the existing approach.

File: nix/deps/nodejs-patched/default.nix [hunk 1]
```diff
@@ -66,6 +67,12 @@ stdenv.mkDerivation {
   patchJavaPhase = ''
     ${nodejs}/bin/node ./node_modules/jetifier/bin/jetify
   '';
+  # Patch node modules using `patch-package`
```
Comment (jakubgs) [inline nix/deps/nodejs-patched/default.nix]: This comment is inaccurate.

Also, you should have the comment explain WHY something is being done, not WHAT, we can see WHAT is done.
A correct comment would be something like:
```suggestion
  # FIXME: Remove this once release newer than 1.19.0 is used which includes:
  # https://github.com/facebook/yoga/commit/f174de70
```

**Review by jakubgs — APPROVED** [review]
Nice.

**Review by siddarthkay — APPROVED** [review]

**Pull request merged**